### PR TITLE
feature: section numbering

### LINKED ISSUE #58: Turning auto-numbering off
Placing `:numbered!:` at the top of the file does not turn auto-numbering of sections off.

PS.  Atom-asciidoc-preview rocks!  I use it  every day!!

### PR DESCRIPTION
**Pull request by ldez**
# Description

Rethink the section numbering option:

- soft on: Turn it on unless the document turns it off (`-a sectnums=@`) **Default**
- Always on (`-a sectnums`)
- Always off (`-a sectnums!`)
- Let the document decide

## Screenshots

capture du 2016-06-01 02-17-42

Fix  #58

### CODE REVIEW

**Comment by ldez** [comment]
New menu:

menu

**Comment by mojavelinux** [comment]
I think you pulled the trigger a little bit early on this one. I was going to suggestion shorter label names. While the labels I provided was the right description, they weren't the right fit for the UI, IMO.

I'll propose two groups:

* Set
* Set (soft)
* Unset
* Not set

This terminology conforms best with the user manual. However, I realize that it may not read as well in this context. Therefore, I'll propose a second group:

* On
* Preferred
* Off
* Unspecified

I really like the word "preferred" because it clearly expresses that you would like it to be set if possible, but the document has the final say. I think this second group is the phrasing that fits best.

**Comment by ldez** [comment]
The first group is not user friendly.

In the second group:
- `preferred` is unclear for me.
- `on` or `off` don't tell that option override the document option.

**Comment by mojavelinux** [comment]
I agreed the first group isn't the right wording. I'm in favor of the
second group.

I think we can assume the menu options will be documented fully elsewhere.
The goal of a menu item isn't to provide all the details, but to be
distinct from its neighbors.

It's too wordy to say "let the document decide" since we'll need to repeat
this for every setting.

I'll keep thinking about a better word for preferred. Perhaps "Suggest" or
"Recommend".

**Comment by mojavelinux** [comment]
I meant "Suggested" rather than "Suggest"

**Comment by ldez** [comment]
I think a user interface must be used without documentation.

I propose:

- Always on
- Preferred (unclear for me but if you prefer)
- Always off
- Document

**Comment by mojavelinux** [comment]
I did some driving and I came up with a new proposal:

* Always enabled
* Enabled by default
* Always disabled
* Not specified (or perhaps "Default")

I really like "enabled by default" is what we were trying to say with "preferred" or "suggested". It's a new default, but something that can be changed by the document.

The last one is the toughest. It's actually incorrect to say that the document decided, because the processor itself my decide if the document doesn't specify. So what we're really talking about is letting the default behavior play out.

**Comment by ldez** [comment]
see #164
